client/app/components/dashboards/dashboard-widget/TextboxWidget.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { markdown } from 'markdown';
+import Dropdown from 'antd/lib/dropdown';
+import Menu from 'antd/lib/menu';
+import Modal from 'antd/lib/modal';
+import HtmlContent from '@/components/HtmlContent';
+import TextboxDialog from '@/components/dashboards/TextboxDialog';
+
+function TextboxWidget({ widget, showDropdown, showDeleteButton, onDelete }) {
+  const [text, setText] = useState(widget.text);
+
+  const editTextBox = () => {
+    TextboxDialog.showModal({
+      text: widget.text,
+      onConfirm: (newText) => {
+        widget.text = newText;
+        setText(newText);
+        return widget.save();
+      },
+    });
+  };
+
+  const deleteTextbox = () => {
+    Modal.confirm({
+      title: 'Delete Textbox',
+      content: 'Are you sure you want to remove this textbox from the dashboard?',
+      okText: 'Delete',
+      okType: 'danger',
+      onOk: () => widget.delete().then(onDelete),
+      maskClosable: true,
+      autoFocusButton: null,
+    });
+  };
```

Comment:
I was thinking that `delete` should be implemented in DashboardWidget only.
The only props that need to be overridden by `TextboxWidget` and `VizWidget` would be `title` and `content` of the `Modal`.

Same goes the "Remove from Dashboard" menu item - can be appended by `DashboardWidget` to both subclasses. Wdyt?